aptos-move/framework/aptos-names/sources/verify.move

```diff
@@ -0,0 +1,32 @@
+module aptos_names::verify {
+    use std::string;
+    use aptos_framework::chain_id;
+    use aptos_std::ed25519;
+    use aptos_names::config;
+
+    friend aptos_names::domains;
+
+    struct RegisterDomainProofChallenge has drop {
+        sequence_number: u64,
+        register_address: address,
+        domain_name: string::String,
+        chain_id: u8,
+    }
+
+    const EINVALID_PROOF_OF_KNOWLEDGE: u64 = 1;
+
+    public(friend) fun assert_register_domain_signature_verifies(signature: vector<u8>, sequence_number: u64, account_address: address, domain_name: string::String) {
+        let chain_id = chain_id::get();
+        let register_domain_proof_challenge = RegisterDomainProofChallenge {
+            sequence_number,
+            register_address: account_address,
+            domain_name,
+            chain_id
+        };
+
+        let admin_public_key_bytes = config::admin_public_key();
+        let admin_public_key = ed25519::new_unvalidated_public_key_from_bytes(admin_public_key_bytes);
```

Comment:
Isn't the admin public key a multisig key (multied25519)?

Reply:
As discussed offline, this will not be the same as the admin key. We should name this something else to avoid confusion.

Reply:
yup will update